python/lsst/analysis/drp/plotUtils.py

```diff
@@ -105,3 +107,163 @@ def addPlotInfo(fig, plotInfo):
+            The maximum y edge of the box to use for initial fitting
+        mHW : `float`
+            The hardwired gradient for the fit
+        bHW : `float`
+            The hardwired intercept of the fit
+        mODR : `float`
+            The gradient calculated by the ODR fit
+        bODR : `float`
+            The intercept calculated by the ODR fit
+        yBoxMin : `float`
+            The y value of the fitted line at xMin
+        yBoxMax : `float`
+            The y value of the fitted line at xMax
+        bPerpMin : `float`
+            The intercept of the perpendicular line that goes through xMin
+        bPerpMax : `float`
+            The intercept of the perpendicular line that goes through xMax
+        mODR2 : `float`
+            The gradient from the second round of fitting
+        bODR2 : `float`
+            The intercept from the second round of fitting
+        mPerp : `float`
+            The gradient of the line perpendicular to the line from the
+            second fit
+
+    Notes
+    -----
+    The code does two rounds of fitting, the first is initiated using the
+    hardwired values given in the `paramDict` parameter and is done using
+    an ODR fit to the points defined by the box of xMin, xMax, yMin and
+    yMax. Once this fitting has been done a perpendicular bisector is
+    calculated at either end of the line and only points that fall within
+    these lines are used to recalculate the fit.
+    """
+
+    # Points to use for the fit
+    fitPoints = np.where((xs > paramDict["xMin"]) & (xs < paramDict["xMax"])
+                         & (ys > paramDict["yMin"]) & (ys < paramDict["yMax"]))[0]
+
+    def linearFit(B, x):
+        return B[0]*x + B[1]
+
+    linear = scipyODR.Model(linearFit)
+    linear = scipyODR.polynomial(1)
+
+    data = scipyODR.Data(xs[fitPoints], ys[fitPoints])
+    odr = scipyODR.ODR(data, linear, beta0=[paramDict["bHW"], paramDict["mHW"]])
+    params = odr.run()
+    mODR = float(params.beta[1])
+    bODR = float(params.beta[0])
+
+    paramsOut = {"xMin": paramDict["xMin"], "xMax": paramDict["xMax"], "yMin": paramDict["yMin"],
+                 "yMax": paramDict["yMax"], "mHW": paramDict["mHW"], "bHW": paramDict["bHW"],
+                 "mODR": mODR, "bODR": bODR}
+
+    # Having found the initial fit calculate perpendicular ends
+    mPerp = -1.0/mODR
+    # When the gradient is really steep we need to use
+    # the y limits of the box rather than the x ones
+    if mODR > 5:
+        yBoxMin = paramDict["yMin"]
+        xBoxMin = (yBoxMin - bODR)/mODR
+    else:
+        yBoxMin = mODR*paramDict["xMin"] + bODR
+        xBoxMin = paramDict["xMin"]
+
+    bPerpMin = yBoxMin - mPerp*xBoxMin
+
+    paramsOut["yBoxMin"] = yBoxMin
+    paramsOut["bPerpMin"] = bPerpMin
+
+    if mODR > 5:
+        yBoxMax = paramDict["yMax"]
+        xBoxMax = (yBoxMax - bODR)/mODR
+    else:
+        yBoxMax = mODR*paramDict["xMax"] + bODR
+        xBoxMax = paramDict["xMax"]
```

Comment:
For simplicity/readability, you could combine these into a single if/else statement, instead of two separate ones.